src/contexts/watchlater.context.js

```diff
@@ -0,0 +1,73 @@
+import axios from "axios";
+import { createContext, useContext, useReducer } from "react";
+import { useAuth } from "./";
+import { watchlaterReducer } from '../reducers'
+
+const WatchlaterContext = createContext()
+
+export const WatchlaterProvider = ({ children }) => {
+    const [watchlaterState, watchlaterDispatch] = useReducer(watchlaterReducer, {
+        watchlaterVideos: [],
+        loading: false,
+    })
+    const { getUserToken } = useAuth()
+
+    async function getWatchlater() {
+        watchlaterDispatch({ type: 'SET_LOADING' })
+        try {
+            const response = await axios.get('/api/user/watchlater', {
+                headers: {
+                    authorization: getUserToken()
+                }
+            })
+            return response.data.watchlater
+        } catch (e) {
+            return e.response.status
+        } finally {
+            watchlaterDispatch({ type: 'REMOVE_LOADING' })
+        }
+    }
+
+    async function addToWatchlater(video) {
+        try {
+            const response = await axios.post('/api/user/watchlater', {
+                video
+            }, {
+                headers: {
+                    authorization: getUserToken()
+                }
+            })
+            return response.data.watchlater
+        } catch (e) {
+            return e.response.status
+        }
+    }
+
+    async function removeFromWatchlater(videoId) {
+        try {
+            await axios.delete(`/api/user/watchlater/${videoId}`, {
+                headers: {
+                    authorization: getUserToken()
+                }
+            })
```

Comment:
I'm not returning the response as I don't need the whole array to update state. I'm dispatching the `_id` of video to filter that particular video from the array. I'm only return response in case of any error to to check error codes and display the respective alert.